streams/src/test/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdaterTest.java

```diff
@@ -603,6 +655,178 @@ public void shouldDrainFailedTasksAndExceptions() throws Exception {
         verifyDrainingExceptionsAndFailedTasks(expectedExceptionAndTasks2, expectedExceptionAndTasks3, expectedExceptionAndTasks4);
     }
 
+    @Test
+    public void shouldGetTasksFromInputQueue() {
+        stateUpdater.shutdown(Duration.ofMillis(Long.MAX_VALUE));
```

Comment:
I had another thought/comment about this: what if we just rely on `start()` of the DefaultStateUpdater (I know it means in the production code we'd need to cast to DefaultStateUpdater and call `start()` upon initialization), and then we would not need the extra boolean, and also do not need to lazy-create the thread upon `add`.

Reply:
I added `start()` to the interface. No casting needed.